src/events/screen_view_event.cpp

```diff
@@ -0,0 +1,39 @@
+/*
+Copyright (c) 2022 Snowplow Analytics Ltd. All rights reserved.
+
+This program is licensed to you under the Apache License Version 2.0,
+and you may not use this file except in compliance with the Apache License Version 2.0.
+You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the Apache License Version 2.0 is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+*/
+
+#include "screen_view_event.hpp"
+
+using namespace snowplow;
+using std::invalid_argument;
+
+ScreenViewEvent::ScreenViewEvent() {
+  this->id = NULL;
+  this->name = NULL;
+}
+
+EventPayload ScreenViewEvent::get_custom_event_payload(bool use_base64) const {
+  if (name == NULL && id == NULL) {
+    throw invalid_argument("Either name or id field must be set");
+  }
+
+  json data;
+  if (id != NULL) {
+    data[SNOWPLOW_SV_ID] = *id;
+  }
+  if (name != NULL) {
+    data[SNOWPLOW_SV_NAME] = *name;
+  }
+
+  SelfDescribingJson event = SelfDescribingJson(SNOWPLOW_SCHEMA_SCREEN_VIEW, data);
```

Comment:
Having dedicate classes for the events is fantastic!!
Do you think we could also move the constants for schema and params in the event classes?
That would be helpful for the automatic event class generation

Also, I don't know why we have two different screen_view schemas, one for mobile and one for the rest.
We should probably consider to adopt just one as it would make the life easier for the data modelling.
Btw, another thing I'll add on Discussions.

Reply:
Good point, I moved the schema constants to header files of self-describing events.

I agree that we should at least make it clear that the two screen views are different, maybe call one the DesktopScreenView and the other the MobileScreenView. Or even better deprecate one of them and use the same across all trackers. We can discuss in the Discussions.